civihr_employee_portal/js/reports.js

```diff
@@ -926,6 +926,74 @@
             }.bind(this)
         });
 
+        $.ajax({
+            type: 'GET',
+            url: this.getJsonUrl() + ( typeof path !== 'undefined' ? path : '' ),
+            success: function(data) {
+                var mappedResults = normaliseSummaryData(data.results);
+
+                replaceSummaryTable(mappedResults, this.$DOM);
+            }.bind(this)
+        });
+
+        function normaliseSummaryData(summary) {
+            var normalisedSummary = {},
+                i;
+
+            // The data can have two formats, depending on particular report:
+            // data = { results: [{ departament: "xxx", count: 1 }, ...]) }
+            // or
+            // data = { results: [{ gender: "aaa" }, { gender: "aaa" }, ...] }
+            //
+            // The code inside if takes care of the first possibility
+            // The else takes care of the second
+            if (summary[0] !== undefined && summary[0].data.count !== undefined) {
```

Comment:
What if summary[0] is `undefined` because there is no actual data returned? Does your code handles this scenario?